entrypoint.sh

```diff
@@ -24,49 +24,29 @@
 #
 # Entrypoint for CMS Ansible Execution Environment (aee)
 
-PREVIOUS_LAYER_DIR=/inventory/layer${LAYER_PREVIOUS}
-CURRENT_LAYER_DIR=/inventory/layer${LAYER_CURRENT}
-
-create_complete() {
-  mkdir -p ${CURRENT_LAYER_DIR}
-  echo $1 > ${CURRENT_LAYER_DIR}/complete
-}
-
-wait_for_previous() {
-  PREVIOUS_FLAG=${PREVIOUS_LAYER_DIR}/complete
-  until [ -f ${PREVIOUS_FLAG} ]
-  do
-       sleep 5
-       echo "Waiting for the previous configuration layer to complete"
-  done
-  echo "Previous layer completed"
-  COMPLETE=$(cat $PREVIOUS_FLAG)
-  if [ $COMPLETE -ne 0 ]; then
-      echo "The previous configuration layer encountered an error, so this layer will not be run.";
-  fi
-  return $COMPLETE
-}
-
 /entrypoint-setup.sh
 
 if [ $? -ne 0 ]; then
-    create_complete 1
-    exit;
-fi
-
-if [ ! -z ${LAYER_PREVIOUS:+x} ]; then
-  wait_for_previous
-fi
-
-if [ $? -ne 0 ]; then
-    create_complete 1
-    exit;
+    exit 1;
 fi
 
-echo "Running $SESSION_PLAYBOOK from repo $SESSION_CLONE_URL"
-
-"$@" 2>&1
-ANSIBLE_EXIT=$?
-
-create_complete $ANSIBLE_EXIT
-exit $ANSIBLE_EXIT
+ANSIBLE_DIR=/etc/ansible
+
+for layer in $(echo "${@}" | jq -c .[]); do
+    export SESSION_CLONE_URL=$(echo "${layer}" | jq -r .cloneUrl)
+    export SESSION_PLAYBOOK=$(echo "${layer}" | jq -r .playbook)
+    export LAYER_CURRENT=$(echo "${layer}" | jq -r .layer)
+    LAYER_DIR=${ANSIBLE_DIR}/layer${LAYER_CURRENT}
+    PLAYBOOK_PATH=${LAYER_DIR}/${SESSION_PLAYBOOK}
+    export ANSIBLE_ROLES_PATH=${LAYER_DIR}/roles
+
+    echo "Running $SESSION_PLAYBOOK from repo $SESSION_CLONE_URL"
+    ansible-playbook $PLAYBOOK_PATH $ANSIBLE_ARGS
```

Comment:
Where does ANSIBLE_ARGS get set?

Reply:
It's set in the container spec.  See the cfs-operator change.